runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -263,6 +263,8 @@ TR_RelocationRuntime::prepareRelocateAOTCodeAndData(J9VMThread* vmThread,
    TR_ASSERT(_options, "Options were not correctly initialized.");
    _reloLogger->setupOptions(_options);
 
+   TR_RelocationRuntime::IsLoading startLoading(this);
```

Comment:
There are 4 places where `prepareRelocateAOTCodeAndData()` is called from:
1. `installAOTCachedMethod()` when we perform an AOT load
2. `compilationEnd()` when we relocate at the end of an AOT compilation
3. `remoteCompilationEnd()` for relocating at the end of a non-AOT remote compilation
4. `remoteCompilationEnd()` for relocating at the end of an AOT remote compilation

Before this change, the `_isLoading` flag was set only during an AOT (case 1). 
After this change, the `_isLoading` flag is going to be set in all cases. Is this going to have some implications?
I see it used here:
```
void TR_RelocationRuntime::addClazzRecord(uint8_t *ia, uint32_t bcIndex, TR_OpaqueMethodBlock *method)
   {
   if (getPersistentInfo()->isRuntimeInstrumentationEnabled() && _isLoading)
      {
      comp()->addHWPBCMap(_compInfo->getHWProfiler()->createBCMap(ia, bcIndex, method, comp()));
      }
   }
```
It's only relevant for RuntimeInstrumentation, so problems (if any) will appear only in this mode.
We could easily disable RI during JITServer mode (this scenario is something I have not taken into consideration), but what about an AOT compilation when RI is enabled?

Reply:
I am ok with this, but we also need a big comment explaining why those conditions are needed

Reply:
I also now realize that `_isLoading` was originally intended to be exactly that, a flag that the runtime was loading a method from the local SCC and not relocating it in the other scenarios, hence the name. Maybe it's not necessary to test for `_isLoading` in that method any more, in that case.

Reply:
Maybe we should leave the original flag `_isLoading` like it was before, and add another flag `_isRelocating` (or something like that) that is set during `prepareRelocateAOTCodeAndData()`

Reply:
I've restored the original `_isLoading` behaviour and added `_isRelocating` and related functions instead.